kong/plugins/apifortress/fortress_serializer.lua

```diff
@@ -0,0 +1,99 @@
+-- ©2016 API Fortress Inc.
+
+local _M = {}
+function string.starts(String,Start)
+   return string.sub(String,1,string.len(Start))==Start
+end
+
+function string.ends(String,End)
+   return End=='' or string.sub(String,-string.len(End))==End
+end
+
+function _M.serialize(ngx)
+  local authenticated_entity
+  if ngx.ctx.authenticated_credential ~= nil then
+    authenticated_entity = {
+      id = ngx.ctx.authenticated_credential.id,
+      consumer_id = ngx.ctx.authenticated_credential.consumer_id
+    }
+  end
+
+	local contentType =  nil
+	local request_headers = {}
+	local reqBody = ngx.req.get_body_data()
+
+	local postBody = nil
+	local putBody = nil
+	local method = ngx.req.get_method()
+	local reqContentType = nil
+	if method=="POST" then
+		postBody = reqBody
+	end
+	if method=="PUT" or method=="PATCH" then
+		putBody = reqBody
+	end
+	for k, v in pairs(ngx.req.get_headers()) do
+		local val = v
+		if type(val)=="table" then val = val[0] end
+		local item = {name=k,value=v}
+		if k=="content-type" then
+			reqContentType = v
+		end
+		table.insert(request_headers,item)
+	end
+	local response_headers = {}
+	local propCompressed = false
+	for k,v in pairs(ngx.resp.get_headers()) do
+		if type(val)=="table" then val = val[0] end
+		local item = {name=k,value=v}
+		if k=="content-type" then
+			contentType = v
+		end
+		if k=='content-encoding' and v=='gzip' then
+			propCompressed = true
+		end
+		table.insert(response_headers,item)
```

Comment:
This would offer better performance:
```
response_headers[#response_headers+1] = item 
```

If you wish to stick with `table.insert`, the "caching of globals" practise also applies :)